### PR TITLE
fix: [WIP BUG] Add ProfilesDir & DevicesDir to local config

### PR DESCRIPTION
**Pull request by jim-wang-intel**
This is so that the values can be overriden with Envs

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-mqtt-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by lenny-intel — APPROVED** [review]
LGTM

File: cmd/res/configuration.yaml [hunk 1]
```diff
@@ -57,3 +57,8 @@ MQTTBrokerInfo:
   Writable:
     # ResponseFetchInterval specifies the retry interval(milliseconds) to fetch the command response from the MQTT broker
     ResponseFetchInterval: 500
+
+Device:
+  # These have common values (currently), but must be here for service local env overrides to apply when customized
+  ProfilesDir: "./res/profiles"
+  DevicesDir: "./res/devices"
```
Comment (lenny-intel) [inline cmd/res/configuration.yaml:64]: Move up above MQTTBrokerInfo  so MQTTBrokerInfo (custom) is always last

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: cmd/res/configuration.yaml (hunk 1 above)
Comment (jim-wang-intel) [reply cmd/res/configuration.yaml]: fixed

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by farshidtz** [comment]
> @farshidtz , the snap testing is failing with the following error:
> 
> ```
> "failed to unmarshall configuration file /var/snap/edgex-device-mqtt/x1/config/device-mqtt/res/configuration.yaml: yaml: unmarshal errors:\n  line 67: mapping key \"Device\" already defined at line 30"
> ```
> 
> Is the snap somehow adding the `Device` section??? This PR is adding it back and the source YAML only has it once. THX!

The tests add the Device section to work around the issue during testing. I'll remove it now that the config is being added locally.

**Review by farshidtz — APPROVED** [review]

**Pull request merged**